api/sendfild/service/service_test.go

```diff
@@ -0,0 +1,646 @@
+package service
+
+import (
+	"context"
+	"math/rand"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/api/apistruct"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log/v2"
+	mh "github.com/multiformats/go-multihash"
+	"github.com/stretchr/testify/require"
+	"github.com/textileio/go-ds-mongo/test"
+	"github.com/textileio/powergate/v2/lotus"
+	"github.com/textileio/powergate/v2/tests"
+	powutil "github.com/textileio/powergate/v2/util"
+	pb "github.com/textileio/textile/v2/api/sendfild/pb"
+	"github.com/textileio/textile/v2/util"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/test/bufconn"
+)
+
```

Comment:
Tried to be very thorough for these tests. There are an unreasonable number of permutations of all the `ListTxns` options, so just tried to test the obvious ones there.